python/revng/pypeline/cli/project/daemon.py

```diff
@@ -0,0 +1,88 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import logging
+
+import click
+import uvicorn
+import yaml
+
+import revng
+from revng.pypeline.daemon.app import app
+from revng.pypeline.daemon.daemon import Daemon
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    "--host",
+    type=str,
+    default="127.0.0.1",
+    help="The interface to bind to.",
+    show_default=True,
+)
+@click.option(
+    "--port",
+    type=int,
+    default=8000,
+    help="The port to bind to.",
+    show_default=True,
+)
+@click.option(
+    "--workers",
+    type=int,
+    default=1,
+    help="Number of worker processes.",
+    show_default=True,
+)
+@click.option(
+    "--debug",
+    is_flag=True,
+    help="Enable debug mode.",
+)
+@click.option(
+    "--reload",
+    is_flag=True,
+    help="Enable auto-reload on file changes.",
+)
+# TODO: expose all other uvicorn parameters from
+# https://github.com/Kludex/uvicorn/blob/master/uvicorn/main.py
```

Comment:
I'm not sure we want to expose all the CLI options of uvicorn here, that means every time an option is changed in unicorn it needs to be changed here. I'd spend a little bit of time to see if there is an easy way to expose an `--uvicorn-args` option that works similarly to the old `revng daemon` (e.g. running main and manually composing the `argv`). Then the important options (e.g. `--port` or `--bind`) can be added as shortcuts